Feature/4 add trashcan register

7zrv opened the pull request:
resolved : #4 

📌 쓰레기통 등록 및 아이디를 통한 단건 조회 기능 구현

Commit 1: 쓰레기통 등록 API 구현

- 쓰레기통 등록 요청을 받아 처리하는 API를 구현했습니다.
- 요청 본문에 포함된 쓰레기통 정보를 바탕으로 데이터베이스에 새로운 쓰레기통 데이터를 생성합니다.
- 유효성 검증은 요청DTO에 등록후 컨트롤러에서 실행되도록 구현했습니다.

Commit 2: id를 통한 쓰레기통 단건조회 API 구현

- 쓰레기통의 id로 단건조회를 하는 API를 구현했습니다.
- 데이터가 없을시 BadRequest 예외를 반환하도록 했고 전역 예외처리 핸들러를 이용했습니다.

✅ PR 포인트 & 궁금한 점
쓰레기통 등록과 단건 조회 API를 구현했습니다.
구현된 API의 유효성 검사 및 예외 처리 방식에 대해 피드백을 받고 싶습니다.

7zrv committed:
feat: 쓰레기통 등록 API 구현

- 쓰레기통 등록을 위한 컨트롤러 엔드포인트 구현
- 입력 검증 및 데이터 전송을 위한 record 타입의 RegisterTrashcanRequestDto 추가
- 요청DTO에 위도, 경도 및 주소 필드에 대한 검증 어노테이션 추가
- 검증 실패에 대한 예외 처리 포함
- 쓰레기통 등록 처리를 위한 TrashcanService 메서드 생성
- 쓰레기통 엔티티 DB 연결을 위한 Repository 생성

7zrv committed:
feat: id를 통한 쓰레기통 조회 API 구현

- 쓰레기통 조회를 위한 컨트롤러 엔드포인트 구현
- 검색 실패에 대한 예외 처리 포함
- 쓰레기통 등록 처리를 위한 TrashcanService 메서드 생성

Leehunil commented on src/main/java/com/sscanner/team/global/exception/ExceptionCode.java:
```diff
@@ -19,7 +20,11 @@ public enum ExceptionCode {
     DUPLICATED_PRODUCT_NAME(409, "이미 존재하는 상품명입니다.");
+    */
 
+    NOT_FOUND_TRASHCAN_ID(400, "해당하는 쓰레기통이 존재하지 않습니다."),
```
저는 NOT_FOUND 는 404에러로 알고 있습니다!!

7zrv replied:
저는 잘못된 요청에 대한 에러 구현이라 400번으로 해봤는데요
훈일님 말씀을 보니 오해의 여지가 있는 예외명 같아서
NOT_FOUND가 아닌 다른 이름으로 바꿔야할 지 내일 같이 의논해보면 좋을것 같습니다!

Leehunil replied:
좋습니다~

Leehunil commented on src/main/java/com/sscanner/team/trashcan/requestDto/RegisterTrashcanRequestDto.java:
```diff
@@ -0,0 +1,47 @@
+
+public record RegisterTrashcanRequestDto(
+
+        @NotNull(message = "위도는 필수입니다.")
```
저는 항상 NotNull, NotEmpty, NotBlank 뭘 사용해야 할지 헷갈리더라고요... 서진님만의 사용 기준이 있는지 궁금합니다

7zrv replied:
https://sanghye.tistory.com/36
위의 글을 보고 필드에 따라 다르게 구분해서 사용하는중입니다 
근데 지금보니까 valid 다시 한번 확인해봐야겠네요 감사합니다 ㅎㅎ

7zrv replied:
NotBlank가 더 적절한 것 같아 수정했습니다

Leehunil replied:
올려주신 글을 보니 저도 명확하게 구분할 수 있었습니다!!

Leehunil commented on src/main/java/com/sscanner/team/trashcan/type/TrashcanStatus.java:
```diff
@@ -0,0 +1,7 @@
+
+public enum TrashcanStatus {
+    EMPTY,
+    intermediate,
```
여기만 소문자로 되어 있는 것 같아요~

7zrv replied:
자세히 봐주셔서 감사합니다 수정하겠습니다

jhk01007 commented:
main이 아니라 develop브랜치로 머지가 되어야 할 것 같습니다!

7zrv committed:
Fix: Enum, Validation 어노테이션 수정

리뷰 해주신 부분들을 수정했습니다.
- TrasgcanStatus의 소문자로 표기된 값 대문자로 수정
- RegisterTrashcanRequestDto의 validation 어노테이션 변경

7zrv committed:
Fix: 예외처리코드 enum 수정

- NOT_FOUND 에러 코드를 NOT_EXIST로 수정
